### ISSUE TITLE
Add find_one method.

### ISSUE THREAD

**Issue by Aaronepower** [utkarshkukreti/select.rs#42]
Currently just to find a single element requires a bit of boilerplate. Consider adding a a `find_one` method that returns `Option<Node>`.

**Comment by utkarshkukreti** [comment]
You could do `.find(predicate).next()`. Or is that what you mean by "a bit of boilerplate"?

**Comment by Aaronepower** [comment]
The boilerplate is not so much in the number of characters as `document.find_one(pred).unwrap()` is close to `document.find(pred).next().unwrap()` it's more the boilerplate of only returning collection types, especially when a lot of them could be single element collections.

**Comment by HighCommander4** [comment]
Actually, it's not quite the same. I'd like `find_one()` to return `None` if there are two or more nodes in the selection.

**Comment by Aaronepower** [comment]
@HighCommander4 I don't think that is intuitive. I've never found myself commonly needing to require that a query result is completely unique. This would be a lot easier to convey through `document.find(pred).size_hint()`/`document.find(pred).collect::<Vec<_>>().len() == 1`
